core/src/main/java/hudson/util/ArgumentListBuilder.java

```diff
@@ -353,7 +353,7 @@ else if (percent && escapeVars
                 percent = (c == '%');
                 if (quoted) quotedArgs.append(c);
             }
-            if(i == 0 && quoted) quotedArgs.insert(0, '"'); else if (i == 0 && !quoted) quotedArgs.append('"');
+            if(i == 0 && quoted) quotedArgs.insert(0, '"'); else if (i == 0) quotedArgs.append('"');
```

Comment:
for better readability:
```suggestion
            if (i == 0 && quoted) {
                if (quoted) {
                	quotedArgs.insert(0, '"'); 
            	} else {
                	quotedArgs.append('"');
            	}
           	}
```

Reply:
@Egor18 , I agree with @MRamonLeon . Since the code is going to be changed, we can improve the readability.